tests/app/dao/notification_dao/test_notification_dao.py

```diff
@@ -570,6 +572,27 @@ def test_get_all_notifications_for_job(sample_job):
     assert len(notifications_from_db) == 5
 
 
+def test_get_latest_sent_notification_for_job_partially_processed_job(sample_job):
```

Comment:
This is a good test but it took me a little while to understand what the test is doing. Would be a bit easier to understand with the data written out in an array:

```
one_s = timedelta(seconds=1)
now = datetime.utcnow()
test_data = [
   (now - 5 * one_s, "sent"),
   (now - 4 * one_s, "sent"),
   (now - 3 * one_s, "sent"),
   (now - 2 * one_s, "pending"),
   (now - 1 * one_s, "pending"),
   (now, "sent"),
]
for updated_at, status in test_data:
   ...

Reply:
Great idea, I'll adjust this test.